src/main.ts

```diff
@@ -43,16 +46,48 @@ async function run(): Promise<void> {
     if (!resp) {
       throw new Error('Unable to authenticate');
     }
-
-    // @ts-expect-error
-    core.setSecret(resp.token);
     // @ts-expect-error
-    core.setOutput('token', resp.token);
+    const installationToken = resp.token;
+
+    // Need to check accessibility if scope set repository
+    if (scope !== '' && scope.split('/').length === 2) {
+      const error = await isExistRepositoryInGitHubApps(
+        installationToken,
+        scope
+      );
+      if (error.error !== '') {
+        throw new Error(error.error);
+      }
+    }
+
+    core.setSecret(installationToken);
+    core.setOutput('token', installationToken);
   } catch (error) {
     if (error instanceof Error) {
       core.setFailed(error.message);
     }
   }
 }
 
+async function isExistRepositoryInGitHubApps(
+  installationToken: string,
+  repository: string
+): Promise<{error: string}> {
+  const installationOctokit = new Octokit({
+    auth: installationToken,
+    baseUrl: process.env.GITHUB_API_URL || 'https://api.github.com',
+  });
+  const accessibleRepositories: listRepositoriesResponse =
+    await installationOctokit.apps.listReposAccessibleToInstallation();
+
+  const repo = accessibleRepositories.data.repositories.find(
+    (item) => item.full_name === repository
+  );
+  if (repo === undefined) {
+    return {error: `GitHub Apps can't accessible repository (${repository})`};
+  }
+
+  return {error: ''};
```

Comment:
Instead of returning an object, let's `throw` an error. This will simplify the error handling logic where we call this function. The exception will get caught by https://github.com/getsentry/action-github-app-token/pull/17/files#diff-4fab5baaca5c14d2de62d8d2fceef376ddddcc8e9509d86cfa5643f51b89ce3dR65

Reply:
i fixed in https://github.com/getsentry/action-github-app-token/pull/17/commits/f318804345d507c11e017643476cacadad34a60f !